system_metrics_collector/test/system_metrics_collector/test_functions.hpp

```diff
@@ -152,26 +152,41 @@ class PromiseSetter
 class MetricsMessageSubscriber : public rclcpp::Node, public PromiseSetter
 {
 public:
-  explicit MetricsMessageSubscriber(const std::string & name)
+  /**
+   *
+   * @param name the node name
+   * @param name the topic name
+   * @param promise_trigger the PromiseSetter::SetPromise will be called when the number of messages
+   * received is >= the promise_trigger.
+   */
+  MetricsMessageSubscriber(const std::string & name, const std::string topic_name)
```

Comment:
`topic_name` is passed by copy. Is this intentional?

Reply:
No, will fix.